spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java

```diff
@@ -156,23 +156,16 @@ public void setResponse(int status, InputStream entity,
 			context.setResponseDataStream(entity);
 		}
 
-		HttpHeaders httpHeaders = new HttpHeaders();
```

Comment:
no need to create HttpHeaders and copy all headers from headers, isOriginResponseGzipped can get from headers directly